engine/src/main/java/org/terasology/engine/entitySystem/event/internal/EventSystemImpl.java

```diff
@@ -83,7 +83,7 @@ public void process() {
     @Override
     public void registerEvent(ResourceUrn uri, Class<? extends Event> eventType) {
         eventIdMap.put(uri, eventType);
-        logger.debug("Registering event {}", eventType.getSimpleName());
+        logger.atDebug().addArgument(() -> eventType.getSimpleName()).log("Registering event {}");
```

Comment:
`getSimpleName()` is most likely O(1) and only called once there when registering events, so impact can be considered neglectable
```suggestion
        logger.debug("Registering event {}", eventType.getSimpleName()); //NOPMD
```